src/test/test-user-util.c

```diff
@@ -48,9 +48,19 @@ static void test_parse_uid(void) {
 
         r = parse_uid("65535", &uid);
         assert_se(r == -ENXIO);
+        assert_se(uid == 100);
+
+        r = parse_uid("0x1234", &uid);
+        assert_se(r == -EINVAL);
+        assert_se(uid == 100);
+
+        r = parse_uid("01234", &uid);
+        assert_se(r == 0);
+        assert_se(uid == 1234);
 
         r = parse_uid("asdsdas", &uid);
         assert_se(r == -EINVAL);
+        assert_se(uid == 1234);
```

Comment:
```suggestion
        r = parse_uid("65535", &uid);
        assert_se(r == -ENXIO);

        r = parse_uid("0x1234", &uid);
        assert_se(r == -EINVAL);

        r = parse_uid("01234", &uid);
        assert_se(r == 0);

        r = parse_uid("asdsdas", &uid);
        assert_se(r == -EINVAL);
```

I find rather confusing testing the `uid` value that remains after the last successful execution. Could we simply check the error if the function is expected to fail?

Reply:
That's on purpose. It is testing that the function does not touch the output variable on failure.

Reply:
👌 Just a last question: would make any sense assigning `UINT32_MAX` to `uid` in case of error instead of leaving it untouched?

Reply:
No. The convention is to not touch the output variable on error. In this case it doesn't matter much either way, but in other cases we make use of this. Having the same convention over all the code base makes the code easier to follow.

For example:
```
  int output = DEFAULT_VALUE;
  r = parse_something(value_from_config_file, &output);
  if (r < 0)
      log_warning("Failed to parse..., ignoring");
 ...
 r = parse_something(value_from_environment_variable, &output);
 if (r < 0)
      log_warning("Failed to parse..., ingorign"):
 ...
 use_value(output);
```

Reply:
:ok_hand: Thank you for the clarification!